flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/SubtaskExecutionAttemptDetailsInfo.java

```diff
@@ -203,7 +203,10 @@ public static SubtaskExecutionAttemptDetailsInfo create(
         final long now = System.currentTimeMillis();
 
         final TaskManagerLocation location = execution.getAssignedResourceLocation();
-        final String locationString = location == null ? "(unassigned)" : location.getHostname();
+        final String locationString =
```

Comment:
I've found that the API handles location in different ways, such as JobExceptionsHandler,JobVertexTaskManagersHandler, they connect the hostname and data port. 
But for SubtasksAllAccumulatorsHandler,SubtasksTimesHandler,SubtaskExecutionAttemptDetailsInfo, they only use the hostname. 

I think it's better to align these behaviors. We can extract the logic of the concat of hostname and port to a method of TaskManagerLocation.

Reply:
Thanks for the comment, I'll align the location generation logic in different places.